tox.ini

```diff
@@ -45,13 +45,55 @@ deps =
     twcurrent: Twisted
     twtrunk: https://github.com/twisted/twisted/archive/trunk.zip
 
+    {trial,coverage}: mock
+
+    coverage: coverage
+
+
+whitelist_externals =
+    coverage: cp
+    coverage: mkdir
+
+
+passenv =
+    PATH
+    LANG LC_ALL
+    CI CONTINUOUS_INTEGRATION
+    TRAVIS TRAVIS_* _system_*
+    CODECOV_OPTIONS
+
+
+setenv =
+    PIP_DISABLE_PIP_VERSION_CHECK=1
+
+    coverage: COVERAGE_FILE={toxworkdir}/coverage/coverage.{envname}
+    {coverage_combine,codecov}: COVERAGE_FILE={toxworkdir}/coverage/coverage
+
+    {coverage,coverage_combine}: COVERAGE_HTML={envlogdir}/coverage_report_html
+    {coverage,coverage_combine,codecov}: COVERAGE_XML={envlogdir}/coverage_report.xml
+
+
 commands =
     "{toxinidir}/.travis/environment"
-    coverage run -p "{envdir}/bin/trial" {posargs:klein}
+
+    # Run trial without coverage
+    trial: trial --random=0 --logfile="{envlogdir}/trial.log" --temp-directory="{envlogdir}/trial.d" {posargs:klein}
+
+    coverage: mkdir -p "{toxworkdir}/coverage"
+    coverage: coverage run --rcfile="{toxinidir}/.coveragerc" "{envdir}/bin/trial" --random=0 --logfile="{envlogdir}/trial.log" --temp-directory="{envlogdir}/trial.d" {posargs:klein}
+
+    # Copy aside coverage data for each test environment in case we want to look at it later
+    coverage: cp "{env:COVERAGE_FILE}" "{envlogdir}/coverage"
+
+    # Run coverage reports, ignore exit status
+    coverage: - coverage html --rcfile="{toxinidir}/.coveragerc" -d "{env:COVERAGE_HTML}"
```

Comment:
Hrm.  It takes pretty consistently a bit under 0.4s on my laptop to run coverage html and alittle over 0.4s to run coverage xml.

The reason it's don't done here after a combine is to make it easy to check coverage for each run separately.  With Klein that doesn't very much (at all, last I checked), so that's not a big deal.

It'll be easy enough to make it separate, so I'll do that.